demo/modal.py

```diff
@@ -0,0 +1,113 @@
+"""Simple modal that looks similar to Angular Component Dialog."""
+
+import mesop as me
+
+
+@me.stateclass
+class State:
+  is_open: bool = False
+
+
+@me.page(
+  security_policy=me.SecurityPolicy(
+    allowed_iframe_parents=["https://google.github.io"]
+  ),
+  path="/modal",
+)
+def app():
+  state = me.state(State)
+
+  with modal(state.is_open):
+    with modal_container():
+      me.text("Delete File", type="headline-5")
+      with me.box():
+        me.text(text="Would you like to delete cat.jpeg?")
+      with modal_actions():
+        me.button("No", on_click=on_click_close_modal)
+        me.button("Yes", on_click=on_click_close_modal)
+
+  with me.box(style=me.Style(padding=me.Padding.all(30))):
+    me.button(
+      "Open Modal", type="flat", color="primary", on_click=on_click_modal_open
+    )
+
+
+def on_click_close_modal(e: me.ClickEvent):
+  state = me.state(State)
+  state.is_open = False
+
+
+def on_click_modal_open(e: me.ClickEvent):
+  state = me.state(State)
+  state.is_open = True
+
+
+@me.content_component
+def modal(is_open: bool):
+  """Renders the background container that will host the modal.
+
+  One current drawback is that it's not possible to close the modal
+  by clicking on the overlay background. This is due to
+  https://github.com/google/mesop/issues/268.
+
+  Args:
+    is_open: Whether the modal is visible or not.
+  """
+  with me.box(
+    style=me.Style(
+      background="rgba(0,0,0,0.4)",
+      display="block" if is_open else "none",
+      height="100%",
+      overflow_x="auto",
+      overflow_y="auto",
+      position="fixed",
+      width="100%",
+      z_index=1000,
+    )
+  ):
+    with me.box(
+      style=me.Style(
+        align_items="center",
+        display="grid",
+        height="100vh",
```

Comment:
I'm a little surprised this is 100vh and the parent box is 100%, I would have expected the opposite